pandas/tests/extension/base/methods.py

```diff
@@ -105,6 +105,13 @@ def test_factorize_equivalence(self, data_for_grouping, na_sentinel):
         tm.assert_numpy_array_equal(l1, l2)
         self.assert_extension_array_equal(u1, u2)
 
+    def test_fillna_copy(self, data_for_fillna):
+        df = pd.DataFrame({"A": data_for_fillna})
+        filled_val = df.iloc[0, 0]
+
+        result = df.fillna(filled_val)
+        assert df.values.base is not result.values.base
```

Comment:
I think that asserting `df.A._values is data_for_fillna` is needed here.

Reply:
Would be good to test this for Series.fillna as well.

Reply:
Thanks for the review. Sure I will. I just want to make sure I am on the right track.